### PR TITLE
Add LWC integration back

### PR DESCRIPTION
**Pull request by animeavi**
The [Modern-LWC](https://github.com/Tsuser1/Modern-LWC) devs kindly added Factions support at my request recently so I decided to dig the old LWC integration code for Factions and make it work again.

I made some changes like clearing barrels as well as chests (made it a string comparison so it's still compatible with 1.13) and moving the config toggles to config.yml.

### CODE REVIEW

**Comment by CmdrKittens** [comment]
I'm unfamiliar with LWC. Can LWC not be used on other item-holding blocks like trapped chests, furnaces, etc?

File: src/main/java/com/massivecraft/factions/zcore/persist/MemoryBoard.java [hunk 1]
```diff
@@ -158,7 +159,18 @@ public void unclaimAllInWorld(String factionId, World world) {
...
+            Iterator<Entry<FLocation, String>> iter = flocationIds.entrySet().iterator();
+            while (iter.hasNext()) {
+                Entry<FLocation, String> entry = iter.next();
+                if (entry.getValue().equals(factionId)) {
+                    LWCFeatures.clearAllChests(entry.getKey());
+                    iter.remove();
+                }
+            }
+        } else {
+            flocationIds.removeFaction(factionId);
```
Comment (CmdrKittens) [inline src/main/java/com/massivecraft/factions/zcore/persist/MemoryBoard.java]: Unless I missed something, I believe running this only in the `else` means that the factionToLandMap will still contain that faction's old places.

**Comment by animeavi** [comment]
Yes, LWC can protect blocks other than chests.
Full list [here](https://github.com/Tsuser1/Modern-LWC/blob/master/src/config/core.yml#L165)

I could add them if needed.

File: src/main/java/com/massivecraft/factions/zcore/persist/MemoryBoard.java (hunk 1 above)
Comment (animeavi) [reply src/main/java/com/massivecraft/factions/zcore/persist/MemoryBoard.java]: To be honest I don't know the answer to this. I mostly added the old code back without much regard.
If you could point me in the right direction I can try to fix this.

File: src/main/java/com/massivecraft/factions/zcore/persist/MemoryBoard.java [hunk 2]
```diff
@@ -165,12 +165,11 @@ public void clean(String factionId) {
...
                 if (entry.getValue().equals(factionId)) {
                     LWCFeatures.clearAllChests(entry.getKey());
-                    iter.remove();
                 }
             }
-        } else {
-            flocationIds.removeFaction(factionId);
         }
+
+        flocationIds.removeFaction(factionId);
```
Comment (animeavi) [inline src/main/java/com/massivecraft/factions/zcore/persist/MemoryBoard.java:172]: @CmdrKittens I believe this is the correct way to do this. Thoughts?

File: modules/Main/pom.xml [hunk 3]
```diff
@@ -428,6 +434,12 @@
             <version>1.3</version>
             <scope>compile</scope>
         </dependency>
+        <dependency>
```
Comment (CmdrKittens) [inline modules/Main/pom.xml:431]: Twice?

File: src/main/java/com/massivecraft/factions/integration/LWCFeatures.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class LWCFeatures {
+    private static LWC lwc;
+
+    public static void setup() {
+        Plugin test = Bukkit.getServer().getPluginManager().getPlugin("LWC");
+        if (test == null || !test.isEnabled()) return;
```
Comment (CmdrKittens) [inline src/main/java/com/massivecraft/factions/integration/LWCFeatures.java]: Maybe swap out the `== null` for `instanceof LWCPlugin` to do both a nullcheck *and* confirm it's not going to somehow result in a cast exception?

File: src/main/java/com/massivecraft/factions/zcore/persist/MemoryBoard.java (hunk 2 above)
Comment (CmdrKittens) [reply src/main/java/com/massivecraft/factions/zcore/persist/MemoryBoard.java:172]: Yeah, this should work out. 👍

**Comment by CmdrKittens** [comment]
> Yes, LWC can protect blocks other than chests.
> Full list [here](https://github.com/Tsuser1/Modern-LWC/blob/master/src/config/core.yml#L165)
> 
> I could add them if needed.

Is there API exposed to get the list of blocks LWC is protecting (since it can be user configured)?

**Comment by animeavi** [comment]
Yes I changed the code to unprotect every block type included in LWC's config now.
I accidentally added LWC twice to pom.xml, yes, removed it in this commit too.

File: src/main/java/com/massivecraft/factions/integration/LWCFeatures.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+        for (int x = 0; x < chests.size(); x++) {
+            if (lwc.findProtection(chests.get(x)) != null) {
+                lwc.findProtection(chests.get(x)).remove();
+            }
+        }
+    }
+
+    private static boolean isProtectedBlock(Block block) {
```
Comment (CmdrKittens) [inline src/main/java/com/massivecraft/factions/integration/LWCFeatures.java]: Perhaps just call this instead, ensuring it uses the same check LWC uses : https://github.com/Tsuser1/Modern-LWC/blob/22d8fa7a343749d07f9edcc9d6df2dfbdccc19ff/src/main/java/com/griefcraft/lwc/LWC.java#L1373-L1389

**Comment by CmdrKittens** [comment]
If you can test out if it all works with that LWC method I mentioned in the comment 15 minutes ago, I think we're good to merge.

**Comment by animeavi** [comment]
> If you can test out if it all works with that LWC method I mentioned in the comment 15 minutes ago, I think we're good to merge.

Good catch, dunno how I missed that.
Changed to it in the latest commit. I did test it and it works just as well, if not better.

**Comment by CmdrKittens** [comment]
Awesome, thanks. I'll look over and pull tonight.

**Comment by CmdrKittens** [comment]
Check out build 151, I slightly re-arranged the code to reduce duplication, and renamed the class to match other integrations.

**Comment by animeavi** [comment]
That's absolutely fine, I was using the name of the old code that used to be in Factions anyway.

**Pull request merged**